Model/Config.php

```diff
@@ -440,4 +441,12 @@ public function isOrderStatusChangedAdminEmailEnabled(): bool
     {
         return $this->scopeConfig->isSetFlag(self::XML_CONFIG_PATH_ORDER_STATUS_CHANGED_ADMIN_EMAIL, ScopeInterface::SCOPE_WEBSITE);
     }
+
+    /**
+     * @return string
+     */
+    public function getCurrencyCode(): string
+    {
+        return (string)$this->scopeConfig->getValue(self::XML_CONFIG_PATH_CURRENCY_CODE, ScopeInterface::SCOPE_WEBSITE);
```

Comment:
Instead of using our own config, can't we use the currency provided by the store or session instead? That way we should always have the right currency for the scope we are in